Sources/AEPOptimize/Optimize.swift

```diff
@@ -149,16 +177,93 @@ public class Optimize: NSObject, Extension {
             eventData[OptimizeConstants.JsonKeys.DATA] = data
         }
 
+        // Add the flag to request sendCompletion
+        eventData[OptimizeConstants.JsonKeys.REQUEST] = [
+            OptimizeConstants.JsonKeys.REQUEST_SEND_COMPLETION: true
+        ]
+
         // Add override datasetId
         if let datasetId = configSharedState[OptimizeConstants.Configuration.OPTIMIZE_OVERRIDE_DATASET_ID] as? String {
             eventData[OptimizeConstants.JsonKeys.DATASET_ID] = datasetId
         }
 
-        let event = Event(name: OptimizeConstants.EventNames.EDGE_PERSONALIZATION_REQUEST,
-                          type: EventType.edge,
-                          source: EventSource.requestContent,
-                          data: eventData)
-        dispatch(event: event)
+        let edgeEvent = event.createChainedEvent(name: OptimizeConstants.EventNames.EDGE_PERSONALIZATION_REQUEST,
+                                                 type: EventType.edge,
+                                                 source: EventSource.requestContent,
+                                                 data: eventData)
+
+        // Increase timeout to 5s to ensure edge requests have enough time to complete.
+        MobileCore.dispatch(event: edgeEvent, timeout: 5) { responseEvent in
+            guard
+                let responseEvent = responseEvent,
+                let requestEventId = responseEvent.requestEventId
+            else {
+                // response event failed or timed out, remove this event's ID from the requested event IDs dictionary and kick-off queue.
+                self.updateRequestEventIdsInProgress.removeValue(forKey: edgeEvent.id.uuidString)
+                self.eventsQueue.start()
+                return
+            }
+
+            let updateCompleteEvent = responseEvent.createChainedEvent(name: OptimizeConstants.EventNames.OPTIMIZE_UPDATE_COMPLETE,
+                                                                       type: EventType.optimize,
+                                                                       source: EventSource.contentComplete,
+                                                                       data: [
+                                                                           OptimizeConstants.EventDataKeys.COMPLETED_UPDATE_EVENT_ID: requestEventId
+                                                                       ])
+            self.dispatch(event: updateCompleteEvent)
+        }
+
+        // In AEP Response Event handle, `requestEventId` corresponds to the UUID for the Edge request.
+        // Storing the request event UUID to compare and process only the anticipated response in the extension.
+        updateRequestEventIdsInProgress[edgeEvent.id.uuidString] = validDecisionScopes
+
+        // add the Edge event to update propositions in the events queue.
+        eventsQueue.add(edgeEvent)
```

Comment:
makes more sense to the reader to move these above the `MobileCore.dispatch` call i think.

particularly, i notice manipulating the `updateRequestEventIdsInProgress` dictionary in the callback, but we hadn't seen any entry made to that dictionary at that point.

Reply:
Yeah, can move those above but I believe these will almost always be executed before the async callback is invoked.